sympy/functions/elementary/trigonometric.py

```diff
@@ -2222,7 +2222,7 @@ class asec(InverseTrigonometricFunction):
     @classmethod
     def eval(cls, arg):
         if arg.is_Number:
-            if arg is S.NaN:
+            if arg in [S.NaN, S.Zero]:
```

Comment:
Yeah, the `asec` will get evaluated then. So once again, no test will be needed for its assumption.